### ISSUE TITLE
Tenant declaration delete /Common/Shared from AS3 Declaration

### ISSUE THREAD

**Issue by BenEkpy** [F5Networks/f5-appsvcs-extension#443]
### Environment
 * Application Services Version: 3.24
 * BIG-IP Version: 15.1

### Summary
Application /Common/Shared disappear in AS3 Declaration after declaring a tenant that use object within /Common/Shared

### Steps To Reproduce
1. Declare pool in /Common/Shared

```json
POST /mgmt/shared/appsvcs/declare

{
		"class": "ADC",
		"schemaVersion": "3.24.0",
		"updateMode": "selective",
		"Common": {
			"class": "Tenant",
			"Shared": { 
				"class": "Application",
				"template": "shared",
				"my_pool": {
					"class": "Pool",
					"members": [
						{
							"servicePort": 80,
							"serverAddresses": [
								"10.10.10.10"
							]
						}
					]
				}
			}
		}
}
```

2. Check the presence of the /Common/Shared application and the pool 

```json

GET /mgmt/shared/appsvcs/declare

```

3. Declare application using pool in /Common/Shared
```json
{
	"class": "ADC",
	"schemaVersion": "3.24.0",
	"updateMode": "selective",
	"my_tenant": {
		"class": "Tenant",
		"my_app": {
			"class": "Application",
			"template": "generic",
			"my_vs": {
				"virtualAddresses": [
					"2.2.2.2"
				],
				"virtualPort": 80,
				"class": "Service_HTTP",
				"pool": {
					"use": "/Common/Shared/my_pool"
				}
			}
		}
	}
}
```

4. Check that the /Common/Shared is now absent

```json

GET /mgmt/shared/appsvcs/declare

```

5. Check the F5 GUI, the pool is still available ( but not in the AS3 declaration)

6. Retry to declare application using pool in /Common/Shared
```json
{
	"class": "ADC",
	"schemaVersion": "3.24.0",
	"updateMode": "selective",
	"my_tenant": {
		"class": "Tenant",
		"my_app": {
			"class": "Application",
			"template": "generic",
			"my_vs": {
				"virtualAddresses": [
					"2.2.2.2"
				],
				"virtualPort": 80,
				"class": "Service_HTTP",
				"pool": {
					"use": "/Common/Shared/my_pool"
				}
			}
		}
	}
}
```

7. Observe the error 
```json
{
  "code": 422,
  "errors": [
    "/my_tenant/my_app/my_vs/pool/use: contains path to non-existent object Common"
  ],
  "declarationFullId": "",
  "message": "declaration is invalid"
}
```

### Expected Behavior
When declaring Tenant, /Common/Shared shouldn't be deleted nor modifier

**Comment by dstokesf5** [comment]
Confirmed, something is wrong here. I have added AUTOTOOL-2420 to our internal product backlog to resolve this.

Are you seeing an error on step 3? I am seeing an error message with configuring /Common on step 3 that could explain why /Common disappeared.

**Comment by BenEkpy** [comment]
I don't see error message on step 3.

Here is the output after executing AS3 request : 

```
{
  "results": [
    {
      "code": 200,
      "message": "no change",
      "host": "localhost",
      "tenant": "Common",
      "runTime": 1099
    },
    {
      "code": 200,
      "message": "success",
      "lineCount": 22,
      "host": "localhost",
      "tenant": "my_tenant",
      "runTime": 1138
    },
    {
      "code": 200,
      "message": "no change",
      "host": "localhost",
      "tenant": "Common",
      "runTime": 1066
    }
  ],
  "declaration": {
    "class": "ADC",
    "schemaVersion": "3.24.0",
    "updateMode": "selective",
    "my_tenant": {
      "class": "Tenant",
      "my_app": {
        "class": "Application",
        "template": "generic",
        "my_vs": {
          "virtualAddresses": [
            "2.2.2.2"
          ],
          "virtualPort": 80,
          "class": "Service_HTTP",
          "pool": {
            "use": "/Common/Shared/my_pool"
          }
        }
      }
    },
    "id": "autogen_b3d19fd0-0b9c-4958-84f3-a54724f1469e",
    "controls": {
      "archiveTimestamp": "2021-04-15T11:22:38.029Z"
    }
  }
}
```

**Comment by dstokesf5** [comment]
Thank you for the update. I am going to try this with version 3.24.0 to see if I get the same results.

**Comment by dstokesf5** [comment]
Confirmed that the behavior is exactly as you described on 3.24.0. It seems like this use case actually got worse in recent versions. We will make sure to add some additional test cases to cover this better as we work on AUTOTOOL-2420.

**Comment by walterk82** [comment]
I'm looking to have this issue addressed as well.  Thank-You.

**Comment by dstokesf5** [comment]
A developer is currently looking into this. We hope to have it resolved in the 3.29.0 release.

**Comment by DumpySquare** [comment]
I'm running into this now.  subscribing for updates.

**Comment by jfreeland** [comment]
Also having this problem.  Please fix.

**Comment by dstokesf5** [comment]
This issue has been resolved by the 3.29.0 release.
